openshift/release/mirror-upstream-branches.sh

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env bash
+
+# Usage: openshift/release/mirror-upstream-branches.sh
+# This should be run from the basedir of the repo with no arguments
+
+
+set -ex
+readonly TMPDIR=$(mktemp -d knativeEventingBranchingCheckXXXX -p /tmp/)
+
+git fetch upstream --tags
+git fetch openshift --tags
+
+# We need to seed this with a few releases that, otherwise, would make
+# the processing regex less clear with more anomalies
+cat >> "$TMPDIR"/midstream_branches <<EOF
+0.2
+0.3
+EOF
+
+git branch --list -a "upstream/release-1.*" | cut -f3 -d'/' | cut -f2 -d'-' > "$TMPDIR"/upstream_branches
+git branch --list -a "openshift/release-v1.*" | cut -f3 -d'/' | cut -f2 -d'v' | cut -f1,2 -d'.' >> "$TMPDIR"/midstream_branches
+
+sort -o "$TMPDIR"/midstream_branches "$TMPDIR"/midstream_branches
+sort -o "$TMPDIR"/upstream_branches "$TMPDIR"/upstream_branches
+comm -32 "$TMPDIR"/upstream_branches "$TMPDIR"/midstream_branches > "$TMPDIR"/new_branches
+
+UPSTREAM_BRANCH=$(cat "$TMPDIR"/new_branches)
+if [ -z "$UPSTREAM_BRANCH" ]; then
+    echo "no new branch, exiting"
+    exit 0
+fi
+echo "found upstream branch: $UPSTREAM_BRANCH"
+readonly UPSTREAM_TAG="knative-v$UPSTREAM_BRANCH.0"
+readonly MIDSTREAM_BRANCH="release-v$UPSTREAM_BRANCH"
+openshift/release/create-release-branch.sh "$UPSTREAM_TAG" "$MIDSTREAM_BRANCH"
```

Comment:
It's a hard assumption that you call this script only from this very directory. I think it would be a better assumption to just assume that this script is in the _same_ directory as create-release-branch.sh (regardless which directory). Otherwise it is very easy to break.

This is my usual combo that I use to detect the directory that this script is located in, maybe its not needed in its full beauty (e.g. the symlink stuff could be ignored)

```bash
# Dir where this script is located
basedir() {
    # Default is current directory
    local script=${BASH_SOURCE[0]}

    # Resolve symbolic links
    if [ -L $script ]; then
        if readlink -f $script >/dev/null 2>&1; then
            script=$(readlink -f $script)
        elif readlink $script >/dev/null 2>&1; then
            script=$(readlink $script)
        elif realpath $script >/dev/null 2>&1; then
            script=$(realpath $script)
        else
            echo "ERROR: Cannot resolve symbolic link $script"
            exit 1
        fi
    fi

    local dir full_dir
    dir=$(dirname "$script")
    full_dir=$(cd "${dir}/.." && pwd)
    echo "${full_dir}"
}
```
```

Reply:
It's our microuniverse of midstream driven from periodic update-to-head.sh execution. It shouldn't be executed outside of that boundary. I can add at least `$(dirname ${BASH_SOURCE[0]})` magic. :)